components/EditableTitle.tsx

```diff
@@ -0,0 +1,42 @@
+import { TextField } from "@equinor/eds-core-react";
+import { ChangeEvent, KeyboardEvent, useEffect, useState } from "react";
+import styles from "./EditableTitle.module.scss";
+
+export const EditableTitle = (props: {
+  defaultText: string;
+  readOnly?: boolean;
+  onSubmit: (arg0: string) => void;
+}) => {
+  const { defaultText, readOnly, onSubmit } = props;
+  const [text, setText] = useState(defaultText);
+
+  useEffect(() => {
+    setText(defaultText);
+  }, [defaultText]);
```

Comment:
Is this effect necessary?

Reply:
Yes, since title can also be changed in another component